File: Best-Time-0to-Buy-and-Sell-Stock/Best-Time-0to-Buy-and-Sell-Stock.py
prices = [7,1,5,3,6,4]

def maxProfit(prices):
    maxs = 0
    mins = prices[0]
    for i in prices:
        maxs = max(maxs, i - mins)
        mins = min(mins, i)
    return maxs
print(maxProfit(prices))

File: Best-Time-0to-Buy-and-Sell-Stock/Best-Time-0to-Buy-and-Sell-Stock.js
let prices = [7,1,5,3,6,4]

var maxProfit = function(prices) {
    let maxP = 0;
    let minBuy = prices[0];

    for (let sell of prices) {
        maxP = Math.max(maxP, sell - minBuy);
        minBuy = Math.min(minBuy, sell);
    }
    return maxP;
    
};
console.log(maxProfit(prices))